File: Ejemplos/Hilos/Java/HilosBasico.java
public class Hilo extends Thread
{
	public Hilo( String nombre )
	{
		super( nombre );			
	}

	public void run()
	{
		System.out.println( this.getName() );
	}

	public static void main(String[] args) throws InterruptedException 
	{
		Hilo HiloA = new Hilo("Hilo A");
		Hilo HiloB = new Hilo("Hilo B");

		HiloA.start();
		HiloB.start();			

		HiloA.join();
		HiloB.join();
					
		System.out.println( "Fin" );
	}
}	

File: Ejemplos/Hilos/C++/hilos_basico.cpp
#include <thread>
#include <iostream>
#include <string>

//Valor que los hilos imprimirán en pantalla
int valor = 0;

void IncrementarValor( std::string nombre )
{
    std::cout<<"Nombre: "<<nombre<<" - Valor: "<<valor<<std::endl;
}

int main( int argc, char *argv[] )
{
    //Creación de los hilos
    std::thread hilo_a( IncrementarValor, "Hilo A" );
    std::thread hilo_b( IncrementarValor, "Hilo B" );

    //Espera la finalización de los hilos creados
    hilo_a.join();
    hilo_b.join();

    return EXIT_SUCCESS;
}
